File: CASTDecorator/BigOAST/for_node.py
from BigOAST.basic_node import BasicNode


class ForNode(BasicNode):
    def __init__(self):
        super().__init__()

        # self.init = ' '.join(str(x) for x in for_node.init or [])
        # self.term = ' '.join(str(x) for x in for_node.cond or [])
        # self.next = ' '.join(str(x) for x in for_node.next or [])

        self.init = None
        self.term = None
        self.next = None

        # self.children = ' '.join(str(x) for x in for_node.stmt.block_items or [])

        pass

    def to_dect(self):
        d = super().to_dect()
        d.update({'init': self.init})
        d.update({'terminal': self.term})
        d.update({'update': self.next})

        return d

File: ABOAT/src/main/java/BigOAST/ForNode.java
package BigOAST;

public class ForNode extends BasicNode {
	private static final long serialVersionUID = 0L;
	private String variable;
	private String init, terminal, update;

	public String getVariable() {
		return variable;
	}

	public String getInit() {
		return init;
	}

	public String getTerminal() {
		return terminal;
	}

	public String getUpdate() {
		return update;
	}

}
